src/utils/logger.ts

```diff
@@ -0,0 +1,70 @@
+import { Logger } from '@ar.io/sdk';
+import { captureException, captureMessage } from '@sentry/react';
+
+type LogLevel = 'debug' | 'info' | 'warn' | 'error';
+type LogContext = Record<string, any>;
+
+export class SentryLogger extends Logger {
+  private context: LogContext = {};
+  private component: string;
+  private logLevel: LogLevel;
+
+  constructor({
+    component,
+    initialContext = {},
+    logLevel = 'info',
+  }: {
+    component: string;
+    initialContext?: LogContext;
+    logLevel?: LogLevel;
+  }) {
+    super({ level: logLevel });
+    this.component = component;
+    this.context = initialContext;
+    this.logLevel = logLevel;
+  }
+
+  setContext(context: LogContext): SentryLogger {
+    this.context = { ...this.context, ...context };
```

Comment:
Wouldn't spreading properties this way overwrite the common ones of the first's with the second's?

Reply:
Accordingly, consider renaming the function `append/add/mergeContext` since `set` sounds like remove and replace to me.